code/datums/diseases/advance/symptoms/deafness.dm

```diff
@@ -54,3 +59,14 @@
 			else
 				to_chat(M, span_userdanger("Your ears pop and begin ringing loudly!"))
 				ears.deaf = min(20, ears.deaf + 15)
+
+/datum/symptom/deafness/on_stage_change(datum/disease/advance/advanced_disease)
+	. = ..()
+	if(!.)
+		return FALSE
+	var/mob/living/carbon/infected_mob = advanced_disease.affected_mob
+	if(advanced_disease.stage >= 4 && causes_permanent_deafness)
+		ADD_TRAIT(infected_mob, TRAIT_DEAF, DISEASE_TRAIT)
+	else
+		REMOVE_TRAIT(infected_mob, TRAIT_DEAF, DISEASE_TRAIT)
```

Comment:
```suggestion
	if(advanced_disease.stage < 5 || !causes_permanent_deafness)
		REMOVE_TRAIT(infected_mob, TRAIT_DEAF, DISEASE_TRAIT)
```
I'd strongly recommend moving the application of the deafness trait to the ear bursting event and changing this segment of code to the above (perhaps without the stage check).

Reply:
Nice catch.  I took the code from another symptom so that one needs the stage 5 fix as well.